### PR TITLE
i18n(de): add translation for testing-guide

### PR DESCRIPTION
**Pull request by mikelgo**
#### What kind of changes does this PR include?

German translation for testing guide https://docs.astro.build/de/guides/testing/

### CODE REVIEW

File: src/pages/de/guides/testing.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
+---
+layout: ~/layouts/MainLayout.astro
+title: Testen
+description: Eine Einführung in das Testen in Astro
+i18nReady: false
+setup: import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
+---
+Testen hilft dir funktionierenden Astro-Code zu schreiben und zu warten. Astro unterstützt viele beliebte Tools für Unit-Tests, Komponenten-Tests und End-to-End-Tests, darunter Jest, Mocha, Jasmine, Cypress und Playwright. Du kannst sogar Framework-spezifische Testbibliotheken wie React Testing Library installieren, um deine UI-Framework-Komponenten zu testen.
```
Comment (hippotastic) [inline src/pages/de/guides/testing.md:8]:
```suggestion
Testen hilft dir, funktionierenden Astro-Code zu schreiben und zu warten. Astro unterstützt viele beliebte Tools für Unit-Tests, Komponenten-Tests und End-to-End-Tests, darunter Jest, Mocha, Jasmine, Cypress und Playwright. Du kannst sogar Framework-spezifische Testbibliotheken wie React Testing Library installieren, um deine UI-Framework-Komponenten zu testen.
```

File: src/pages/de/guides/testing.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+Test-Frameworks ermöglichen es dir, **Aussagen** oder **Erwartungen** über das Verhalten deines Codes in bestimmten Situationen zu formulieren und diese mit dem tatsächlichen Verhalten deines aktuellen Codes zu vergleichen.
+
+## Playwright
+
+Playwright ist ein End-to-End Test-Framework für moderne Web-Apps. Verwende die Playwright-API in JavaScript oder TypeScript, um deinen Astro-Code auf allen modernen Rendering-Engines, darunter Chromium, WebKit und Firefox, zu testen.
+
+### Installation
+
+Du kannst mit der [VS Code Extension](https://playwright.dev/docs/getting-started-vscode) beginnen und deine Tests ausführen.
```
Comment (hippotastic) [inline src/pages/de/guides/testing.md:18]:
```suggestion
Du kannst mit der [VS Code-Erweiterung](https://playwright.dev/docs/getting-started-vscode) beginnen und deine Tests ausführen.
```

File: src/pages/de/guides/testing.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+Alternativ kannst du Playwright in deinem Astro-Projekt mit dem von dir gewählten Paketmanager installieren. Folge den CLI-Schritten, um Javascript/Typescript, den Namen deines Test Ordners und einen optionalen Github Actions Workflow auszuwählen.
```
Comment (hippotastic) [inline src/pages/de/guides/testing.md:20]:
```suggestion
Alternativ kannst du Playwright in deinem Astro-Projekt mit dem von dir gewählten Paketmanager installieren. Folge den CLI-Schritten, um JavaScript/TypeScript, den Namen deines Test-Ordners und einen optionalen Github Actions-Workflow auszuwählen.
```

File: src/pages/de/guides/testing.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  ```
+  </Fragment>
+  <Fragment slot="yarn">
+  ```shell
+  yarn create playwright
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+### Erstelle deinen ersten Playwright Test
```
Comment (hippotastic) [inline src/pages/de/guides/testing.md:40]:
```suggestion
### Erstelle deinen ersten Playwright-Test
```

File: src/pages/de/guides/testing.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+1. Wähle eine Seite zum Testen aus. Wir verwenden die unten stehende `index.astro`-Seite als Beispiel.
+
+```html title="src/pages/index.astro"
+---
+---
+<html lang="de">
+  <head>
+    <title>Astro ist fantastisch!</title>
+    <meta name='description' content="Beziehe Inhalte von überall und stelle ihn schnell mit Astros next-gen Insel-Architektur bereit." />
```
Comment (hippotastic) [inline src/pages/de/guides/testing.md:50]:
```suggestion
    <meta name='description' content="Beziehe Inhalte von überall und stelle sie schnell mit Astros Insel-Architektur der nächsten Generation bereit." />
```

File: src/pages/de/guides/testing.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+  </head>
+  <body></body>
+</html>
+```
+
+2. Erstelle einen neuen Ordner und füge nachstehende Testdatei in `src/test` hinzu. Kopiere und füge den nachfolgenden Test in deine Datei ein, um zu verifizieren, dass die Metainformationen der Seite korrekt sind. Aktualisiere den `<title>`-Wert entsprechend der Seite, die du testest.
+
+```jsx title="src/test/index.spec.ts" "Astro is awesome!"
```
Comment (hippotastic) [inline src/pages/de/guides/testing.md:58]:
```suggestion
```jsx title="src/test/index.spec.ts" "Astro ist fantastisch!"
```

File: src/pages/de/guides/testing.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+test('Metadaten sind korrekt', async ({ page }) => {
+  await page.goto("http://localhost:3000/");
+
+  await expect(page).toHaveTitle('Astro ist fantastisch!');
+});
+```
+
+:::tip[Eine Basis Url setzen]
```
Comment (hippotastic) [inline src/pages/de/guides/testing.md:66]:
```suggestion
:::tip[Eine Basis-URL setzen]
```

File: src/pages/de/guides/testing.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+Du kannst [`"baseURL": "http://localhost:3000"`](https://playwright.dev/docs/api/class-testoptions#test-options-base-url) in der `playwright.config.ts`-Konfigurationsdatei setzen, um `page.goto("/")` anstatt `page.goto("http://localhost:3000/")` als komfortablere URL zu nutzen.
+:::
+
+### Deine Playwright Tests ausführen
```
Comment (hippotastic) [inline src/pages/de/guides/testing.md:70]:
```suggestion
### Deine Playwright-Tests ausführen 
```

File: src/pages/de/guides/testing.md [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+
+Du kannst einen einzelnen oder mehrere Tests auf einmal ausführen und dabei einen oder mehrere Browser testen. Standardmäßig, werden deine Testergebnisse im Terminal angezeigt. Optional kannst du den HTML Test-Reporter öffnen, um einen vollständigen Bericht anzuzeigen und die Testergebnisse zu filtern.
+
+1. Um unseren Test aus dem vorherigen Beispiel mit der Kommandozeile auszuführen, nutze das `test`-Kommando. Optional kannst du den Dateinamen angeben, um nur den einzelnen Test auszuführen.
+
+```sh
+npx playwright test index.spec.ts
+```
+
+2. Um den vollständigen HTML Testbericht anzusehen, öffne ihn mit folgendem Befehl:
```
Comment (hippotastic) [inline src/pages/de/guides/testing.md:80]:
```suggestion
2. Um den vollständigen HTML-Testbericht anzusehen, öffne ihn mit folgendem Befehl:
```

File: src/pages/de/guides/testing.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+npx playwright show-report
+```
+
+:::tip
+Führe deine Test gegen deinen Produktions-Code aus, um näher an deine live-gehostete Seite heranzukommen.
+:::
+
+#### Fortgeschritten: Einen Entwicklungs-Web-Server während deinen Tests starten 
+
+Du kannst Playwright auch deinen Server starten lassen, wenn du dein Test-Skript mit der [`webServer`](https://playwright.dev/docs/test-advanced#launching-a-development-web-server-during-the-tests) Option in der Playwright Konfigurationsdatei ausführst.
```
Comment (hippotastic) [inline src/pages/de/guides/testing.md:91]:
```suggestion
Du kannst Playwright auch deinen Server starten lassen, wenn du dein Test-Skript mit der [`webServer`](https://playwright.dev/docs/test-advanced#launching-a-development-web-server-during-the-tests)-Option in der Playwright-Konfigurationsdatei ausführst.
```

File: src/pages/de/guides/testing.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+
+Hier ist ein Beispiel der Konfiguration und Kommandos, wenn du Yarn verwendest:
+
+1. Füge ein Test-Skript wie: `"test:e2e": "yarn playwright"` in deine `package.json`-Datei in der Projektwurzel ein.
+
+2. In `playwright.config.ts` füge das `webServier`-Objekt hinzu und aktualisiere den Kommando-Wert in `yarn preview`.
+3.
```
Comment (hippotastic) [inline src/pages/de/guides/testing.md:98]:
```suggestion

```

File: src/pages/de/guides/testing.md [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+    reuseExistingServer: !process.env.CI,
+  },
+  use: {
+    baseURL: 'http://localhost:3000/app/',
+  },
+};
+export default config;
+```
+
+3. Führe `yarn build` und anschließend `yarn test:e2e` aus, um deine Playwright Tests auszuführen.
```
Comment (hippotastic) [inline src/pages/de/guides/testing.md:115]:
```suggestion
3. Führe `yarn build` und anschließend `yarn test:e2e` aus, um deine Playwright-Tests auszuführen.
```

**Review by hippotastic — COMMENTED** [review]
Really high quality translation, thank you! I only found a few grammar mistakes.

Most frequently you are missing the hyphen between compound words. You're in good company though - this is the most common mistake, which is also why it's first in [our translation guide](https://github.com/withastro/docs/tree/main/src/i18n/de#h%C3%A4ufige-fehler). :)

Oh, and it would be great if you could add the translation "Testen" to the navigation bar as well. You need to add an entry for `guides/testing` to [this file](https://github.com/withastro/docs/blob/main/src/i18n/de/nav.ts) to make it work.

Thank you and keep up the good work!

**Comment by mikelgo** [comment]
> Really high quality translation, thank you! I only found a few grammar mistakes.
> 
> Most frequently you are missing the hyphen between compound words. You're in good company though - this is the most common mistake, which is also why it's first in [our translation guide](https://github.com/withastro/docs/tree/main/src/i18n/de#h%C3%A4ufige-fehler). :)
> 
> Oh, and it would be great if you could add the translation "Testen" to the navigation bar as well. You need to add an entry for `guides/testing` to [this file](https://github.com/withastro/docs/blob/main/src/i18n/de/nav.ts) to make it work.
> 
> Thank you and keep up the good work!

Thank u very much for your kind words - appreciate it! I fixed all suggestions and added the navigation item as well.

**Review by hippotastic — APPROVED** [review]
LGTM!

**Pull request merged**